api/src/main/java/org/pmiops/workbench/cohortreview/CohortReviewServiceImpl.java

```diff
@@ -130,7 +140,45 @@ public ParticipantCohortStatus findParticipantCohortStatus(Long cohortReviewId,
         return participantCohortStatus;
     }
 
+    @Override
     public List<ParticipantCohortStatus> findAll(Long cohortReviewId, List<Filter> filterList, PageRequest pageRequest) {
         return participantCohortStatusDao.findAll(cohortReviewId, filterList, pageRequest);
     }
+
+    @Override
+    public ParticipantCohortAnnotation saveParticipantCohortAnnotation(ParticipantCohortAnnotation participantCohortAnnotation) {
+        return participantCohortAnnotationDao.save(participantCohortAnnotation);
+    }
+
+    @Override
+    public CohortAnnotationDefinition findCohortAnnotationDefinition(Long cohortAnnotationDefinitionId) {
+        return cohortAnnotationDefinitionDao.findOne(cohortAnnotationDefinitionId);
+    }
+
+    @Override
+    public void deleteParticipantCohortAnnotation(Long annotationId) {
+        //TODO implement this
+    }
+
+    @Override
+    public ParticipantCohortAnnotation findParticipantCohortAnnotation(Long annotationId) {
```

Comment:
How do you expect this and the other findParticipantCohortAnnotation() to get used?

My guess is we're normally going to wind up calling findParticipantCohortAnnotations() and fetching all annotations for a participant at once...

Reply:
@freemabd1 correct me if I'm wrong but we'll need to find individual annotations during update and for some validation, won't we?

Reply:
I think that ought to be done internal to saveParticipantCohortAnnotation.

Reply:
So the CohortReviewController has two endpoints: getParticipantCohortAnnotation() and getParticipantCohortAnnotations(). Currently we will probably only need getParticipantCohortAnnotations(). I added getParticipantCohortAnnotation() because i thought it might be useful.. hence the findParticipantCohortAnnotation(Long annotationId) in this service. I can remove it if we think it's unnecessary.

Reply:
Also, currently the CohortReviewController.updateParticipantCohortAnnotation() takes a ModifyParticipantCohortAnnotationRequest(which only contains the annotation value) which will require a read then save. I'm not stuck on this.. so we could change to pass the full ParticipantCohortAnnotation object instead.. therefore not needed to read.

Reply:
I wouldn't do that -- I'd change your save method to not require a full annotation, and just take the value instead.

(The thought being: whatever validation we're applying ideally happens inside the service in the save method, so that it gets applied to all callers; and so if we have to do anything like read-modify-write in a transaction we can take care of it there.)

I don't know whether there's a need for an endpoint that fetches a single annotation. If we don't have a need for it probably better to remove it for now.